tchannel/tchannel.py

```diff
@@ -232,6 +236,12 @@ def advertise(self, routers, name=None, timeout=None):
             The timeout (in seconds) for the initial advertise attempt.
             Defaults to 30 seconds.
 
+        :param router_file:
+            The host file that contains the routers information. The file
+            should contain a JSON stringified format of the routers parameter.
+            Either routers or router_file should be provided. If both provided,
```

Comment:
Merging two semantics into one variable seems to be an anti-pattern to me. It is confusing to consumers.

On the other hand, node implementation has two separate parameters as well. It is good to keep xlang interfaces consistent.